src/pybind/mgr/dashboard/controllers/certificate.py

```diff
@@ -0,0 +1,223 @@
+import logging
+from typing import Any, Dict, List, Optional
+
+import cherrypy
+from ceph.deployment.service_spec import ServiceSpec
+
+from ..security import Scope
+from ..services.certificate import build_certificate_status_response, \
+    fetch_certificates_for_service, find_certificate_for_service, \
+    get_daemon_hostnames, process_certificates_for_list
+from ..services.exception import handle_orchestrator_error
+from ..services.orchestrator import OrchClient, OrchFeature
+from ..tools import str_to_bool
+from . import APIDoc, APIRouter, EndpointDoc, ReadPermission, RESTController
+from ._version import APIVersion
+from .orchestrator import raise_if_no_orchestrator
+
+logger = logging.getLogger(__name__)
+
+
+CERTIFICATE_LIST_SCHEMA = [{
+    'cert_name': (str, 'Certificate name'),
+    'scope': (str, 'Certificate scope (SERVICE, HOST, or GLOBAL)'),
+    'signed_by': (str, 'Certificate issuer (user or cephadm)'),
+    'status': (str, 'Certificate status (valid, expiring, expired, invalid)'),
+    'days_to_expiration': (int, 'Days remaining until expiration'),
+    'expiry_date': (str, 'Certificate expiration date'),
+    'issuer': (str, 'Certificate issuer distinguished name'),
+    'common_name': (str, 'Certificate common name (CN)'),
+    'target': (str, 'Certificate target (service name or hostname)'),
+    'subject': (dict, 'Certificate subject details'),
+    'key_type': (str, 'Public key type (RSA, ECC, etc.)'),
+    'key_size': (int, 'Public key size in bits')
+}]
+
+
+@APIRouter('/certificate', Scope.HOSTS)
+@APIDoc("Certificate Management API", "Certificate")
+class Certificate(RESTController):
+
+    @EndpointDoc("List All Certificates",
+                 parameters={
+                     'status': (str, 'Filter by certificate status '
+                                     '(e.g., "expired", "expiring", "valid", "invalid")'),
+                     'scope': (str, 'Filter by certificate scope '
+                                    '(e.g., "SERVICE", "HOST", "GLOBAL")'),
+                     'service_name': (str, 'Filter by certificate name '
+                                      '(e.g., "rgw*")'),
+                     'include_cephadm_signed': (bool, 'Include cephadm-signed certificates '
```

Comment:
Maybe we could have a more detailed comment of what `include_cephadm_signed` actually does. What about:
```suggestion
                     'include_cephadm_signed': (bool, 'Enables cephadm to self-sign certificates for seamless operation'
```

Took it from: https://docs.ceph.com/en/latest/cephadm/certmgr/#self-signed-certificates